packages/talos_agent/src/messaging/kafka.rs

```diff
@@ -137,49 +135,47 @@ impl KafkaConsumer {
-                            let list = group_list.groups();
-                            let mut found = false;
-                            for group_info in list {
-                                if group_info.name() == self.config.group_id {
-                                    log::info!("Detected consumer group: name: {}, state: {}", group_info.name(), group_info.state());
-
-                                    if group_info.state() != "Stable" {
-                                        continue;
-                                    }
-
-                                    found = true;
-                                    break;
-                                }
-                            }
-
-                            if found {
-                                break;
-                            }
-                        }
-                        Err(_) => {
-                            log::error!("Cannot fetch group info for group id '{}'", self.config.group_id);
-                            break;
+        let topic = self.config.certification_topic.as_str();
+        let partition = 0_i32;
+
+        let mut partition_list = TopicPartitionList::new();
+        partition_list.add_partition(topic, partition);
+        // This line is required for seek operation to be successful.
+        partition_list.set_partition_offset(topic, partition, Offset::End).unwrap();
+
+        let rslt_assign = self.consumer.assign(&partition_list);
+        match rslt_assign {
+            Ok(()) => {
+                return match self.get_offset(partition, Duration::from_secs(5)) {
+                    Ok(offset) => {
+                        log::info!("Seeking on partition {} to offset: {:?}", partition, offset);
+                        match self.consumer.seek(topic, partition, offset, Duration::from_secs(5)) {
+                            Ok(()) => Ok(()),
+                            Err(e) => Err(format!("Error when Seeking to offset {:?}. Error: {}", offset, e)),
                         }
                     }
+                    Err(e) => Err(format!("Error fetching offset of partition {}. Error: {})", partition, e)),
+                };
+            }
 
-                    log::info!("Consumer group is not ready yet");
-                    thread::sleep(Duration::from_secs(1));
-                }
+            Err(e) => Err(format!("Error invoking assign('{}', {}. Error: {})", topic, partition, e)),
+        }
+    }
 
-                Ok(())
-            }
-            Err(kafka_error) => {
-                error!("Error when subscribing to topics. {:?}", kafka_error);
-                Err(kafka_error.to_string())
+    fn get_offset(&self, partition: i32, timeout: Duration) -> Result<Offset, String> {
+        let now_ms = OffsetDateTime::now_utc().unix_timestamp() * 1000;
+        let topic = self.config.certification_topic.as_str();
+        match self.consumer.offsets_for_timestamp(now_ms, timeout) {
```

Comment:
Although this works, just wondering do we just use the `highest watermark` as the broker can easily provide the high and low watermark on partition rather than searching for the offsets for a particular time, across all partitions (although we are single partition).

And then also you have the overhead of searching the partition and topic to get the offset. 

vs just using this:-

https://docs.rs/rdkafka/latest/rdkafka/consumer/trait.Consumer.html#tymethod.fetch_watermarks

Reply:
I will try to use watermarks.

Reply:
Done